test/framework/prometheus.go

```diff
@@ -63,6 +63,17 @@ type PromRemoteWriteTestConfig struct {
 	ShouldSuccess      bool
 }
 
+func (f *Framework) MakeBasicPrometheusCRForValidation(ns, name string, spec monitoringv1.PrometheusSpec) *monitoringv1.Prometheus {
```

Comment:
not sure that we need the extra function if it's only called from one place.

Reply:
I would argue that in this case, the test can create the Prometheus object directly without an helper function.